### PR TITLE
infoschema: add affected_rows to the processlist table

### LINKED ISSUE #54486: Show affected_rows in `information_schema.processlist`
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Show affected_rows in `information_schema.processlist`

**Describe alternatives you've considered:**

- Special syntax
- Logs. It is already in expensive_logs.

**Teachability, Documentation, Adoption, Migration Strategy:**

For long-running statements, especially pipelined DML(`@@tidb_dml_type=bulk`), we need a straightforward way to know the progress.

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Issue Number: close #54486

Problem Summary:

We need a straightforward way to know the progress of long-running statements.

### What changed and how does it work?

Add a column `affected_rows` to `information_schema.processlist`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add a column `affected_rows` to `information_schema.processlist`.
```

### CODE REVIEW

File: pkg/infoschema/tables.go [hunk 1]
```diff
@@ -854,6 +854,7 @@ var tableProcesslistCols = []columnInfo{
 	{name: "TxnStart", tp: mysql.TypeVarchar, size: 64, flag: mysql.NotNullFlag, deflt: ""},
 	{name: "RESOURCE_GROUP", tp: mysql.TypeVarchar, size: resourcegroup.MaxGroupNameLength, flag: mysql.NotNullFlag, deflt: ""},
 	{name: "SESSION_ALIAS", tp: mysql.TypeVarchar, size: 64, flag: mysql.NotNullFlag, deflt: ""},
+	{name: "AFFECTED_ROWS", tp: mysql.TypeLonglong, size: 21, flag: mysql.UnsignedFlag},
```
Comment (cfzjywxk) [inline pkg/infoschema/tables.go:857]: Is there compatibility risks for tools like dashboard?

Comment (ekexium) [reply pkg/infoschema/tables.go:857]: I suppose they should not depend on the number of the columns...
@lcwangchao have you met any compatibility problems when adding SESSION_ALIAS?

File: pkg/util/processinfo.go [hunk 2]
```diff
@@ -140,7 +140,15 @@ func (pi *ProcessInfo) ToRow(tz *time.Location) []any {
 			diskConsumed = pi.DiskTracker.BytesConsumed()
 		}
 	}
-	return append(pi.ToRowForShow(true), pi.Digest, bytesConsumed, diskConsumed, pi.txnStartTs(tz), pi.ResourceGroupName, pi.SessionAlias)
+
+	var affectedRows any
+	if pi.StmtCtx != nil {
+		affectedRows = pi.StmtCtx.AffectedRows()
+	} else {
+		affectedRows = nil
```
Comment (lcwangchao) [inline pkg/util/processinfo.go:148]: Seems the `else` branch is needless.

File: pkg/infoschema/tables.go (hunk 1 above)
Comment (lcwangchao) [reply pkg/infoschema/tables.go:857]: I think it's fine. The added row is in the last position. @bb7133 Could you take a look?

Comment (bb7133) [reply pkg/infoschema/tables.go:857]: I believe it is fine for compatibility. However, my concern is about the semantics when it is called 'affected rows': all columns in `PROCESSLIST` are the 'current status' but `AFFECTED_ROWS` means the affected rows by the last statement, which could be confusing.

For example, after executing `INSER T INTO ..`, we have:
```
            ID: 2097158
          USER: root
          HOST: 127.0.0.1:50816
            DB: test
       COMMAND: Sleep
          TIME: 16
         STATE: autocommit
          INFO: NULL
        DIGEST: 04fa858fa491c62d194faec2ab427261cc7998b3f1ccf8f6844febca504cb5e9
           MEM: 4096
          DISK: 0
      TxnStart:
RESOURCE_GROUP: default
 SESSION_ALIAS:
 AFFECTED_ROWS: 2
2 rows in set (0.01 sec)
```

Comment (ekexium) [reply pkg/infoschema/tables.go:857]: How about displaying `0` when `INFO` is null or `len(INFO) == 0`?

Comment (bb7133) [reply pkg/infoschema/tables.go:857]: I'm not sure, to be honest. Maybe we could also consider renaming it 'AFFECTED_ROWS_LAST_STATEMENT'?

What is the expected result of this column for pipelined DMLs?

Comment (ekexium) [reply pkg/infoschema/tables.go:857]: It's introduced to reveal the current progress of the statement during its execution, as a pipelined DML may run for a long time. So 'AFFECTED_ROWS_LAST_STATEMENT' is not a good idea I suppose.

Comment (cfzjywxk) [reply pkg/infoschema/tables.go:857]: Or just rename it "CURRENT_AFFECTED_ROWS" to distinguish them and make it clear.

Comment (ekexium) [reply pkg/infoschema/tables.go]: Sounds good

**Review by cfzjywxk — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**